From 44a05e04ae39a4a32ffd089f7e8c6be2624e0d5b Mon Sep 17 00:00:00 2001
From: Riku Voipio <riku.voipio@linaro.org>
Date: Thu, 20 Aug 2015 10:39:02 +0300
Subject: [PATCH] Remove scratchbox2 support

Since scratchbox2 was removed from Debian, remove this special
case from debootstrap. scratchbox2 has been abandoned upstream,
and even within scratchbox users using this debootstrap option
was rare. Thus I think it makes sense to clean up this option
away.
---
 debootstrap   | 12 +-----------
 debootstrap.8 |  4 +---
 scripts/sid   |  6 ++----
 3 files changed, 4 insertions(+), 18 deletions(-)

diff --git a/debootstrap b/debootstrap
index 86e9c15..70a8e04 100755
--- a/debootstrap
+++ b/debootstrap
@@ -102,7 +102,7 @@ usage()
                              archive
       --variant=X            use variant X of the bootstrap scripts
                              (currently supported variants: buildd, fakechroot,
-                              scratchbox, minbase)
+                              minbase)
       --keyring=K            check Release files against keyring K
       --no-check-gpg         avoid checking Release file signatures
       --force-check-gpg      force checking Release file signatures
@@ -457,16 +457,6 @@ fi
 
 if [ "$TARGET" = "/" ]; then
 	CHROOT_CMD=""
-elif doing_variant scratchbox; then
-	for config in ~/.scratchbox2/*/sb2.config;
-	do
-		export `grep ^SBOX_TARGET_ROOT= $config`
-		if [ "x$SBOX_TARGET_ROOT" = "x$TARGET" ]; then
-			SB2_TARGET=$(basename $(dirname $config))
-		fi
-	done
-	[ "x$SB2_TARGET" != "x" ] || error 1 SBOXTARGETREQ "No scratchbox target configured for $TARGET"
-	CHROOT_CMD="sb2 -eR -t $SB2_TARGET"
 else
 	CHROOT_CMD="chroot $TARGET"
 fi
diff --git a/debootstrap.8 b/debootstrap.8
index 5864148..a7c9ff0 100644
--- a/debootstrap.8
+++ b/debootstrap.8
@@ -71,15 +71,13 @@ or apt, and that it is far better to specify the entire base system than
 rely on this option.
 With this option set, this behaviour is disabled.
 .IP
-.IP "\fB\-\-variant=minbase|buildd|fakechroot|scratchbox\fP"
+.IP "\fB\-\-variant=minbase|buildd|fakechroot\fP"
 Name of the bootstrap script variant to use.
 Currently, the variants supported are minbase, which only includes
 essential packages and apt; buildd, which installs the build-essential
 packages into
 .IR TARGET ;
 and fakechroot, which installs the packages without root privileges.
-Finally there is variant scratchbox, which is for creating targets
-for scratchbox usage.
 The default, with no \fB\-\-variant=X\fP argument, is to create a base
 Debian installation in
 .IR TARGET .
diff --git a/scripts/sid b/scripts/sid
index bf3404f..c2092ed 100644
--- a/scripts/sid
+++ b/scripts/sid
@@ -1,7 +1,7 @@
 mirror_style release
 download_style apt
 finddebs_style from-indices
-variants - buildd fakechroot minbase scratchbox
+variants - buildd fakechroot minbase
 keyring /usr/share/keyrings/debian-archive-keyring.gpg
 
 if doing_variant fakechroot; then
@@ -22,7 +22,7 @@ work_out_debs () {
 		#required="$required $(get_debs Priority: important)"
 		#  ^^ should be getting debconf here somehow maybe
 		base="$(get_debs Priority: important)"
-	elif doing_variant buildd || doing_variant scratchbox; then
+	elif doing_variant buildd; then
 		base="apt build-essential"
 	elif doing_variant minbase; then
 		base="apt"
@@ -85,8 +85,6 @@ second_stage_install () {
 
 	if doing_variant fakechroot; then
 		setup_proc_fakechroot
-	elif doing_variant scratchbox; then
-		true
 	else
 		setup_proc
 		in_target /sbin/ldconfig
-- 
2.1.4

